src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/MemoryPoolIterator.cs

```diff
@@ -4,13 +4,28 @@
 using System;
 using System.Diagnostics;
 using System.Numerics;
+using System.Runtime.CompilerServices;
 using System.Threading;
 
 namespace Microsoft.AspNetCore.Server.Kestrel.Internal.Infrastructure
 {
     public struct MemoryPoolIterator
     {
+        // When this variable is set in KestrelEngine.JitReadonlyConsts it will 
+        // deterministically convert the following readonly static values to jitted consts 
+        // and embed the init code in that start up called function.
+        //
+        // This means when they are next used the have been pre-evaluated and the jit can 
+        // either directly embed them or use them for branch elimiation.
+        internal static bool StaticReadonlysJitted;
+
+        private const ulong _powerOfTwoToHighByte = 0x01020304050607;
+
+        // Convert these returns to jitted consts
+        private static readonly bool IsHardwareAccelerated = Vector.IsHardwareAccelerated;
+        private static readonly bool IsLittleEndian = BitConverter.IsLittleEndian;
         private static readonly int _vectorSpan = Vector<byte>.Count;
+        private static readonly int _vectorUlongSpan = Vector<ulong>.Count;
```

Comment:
Were you not seeing `Vector.IsHardwareAccelerated` and `Vector<xxx>.Count` being replaced by constants during jiting when used in-line in the method?  The code you have now gets them replaced by constants during importing due to an optimization for `ldsfld` on read-only static fields, but `impSIMDIntrinsic` is supposed to kick in and replace their references with constants if you use them in-line.  Doing the latter should have the benefit that the Jit could recognize that this is where the constants came from so that it could e.g. use that for an unrolling heuristic and unroll the loop in `Seek` to use constant rather than variable indexing (dotnet/coreclr#7422).